src/gwt/src/org/rstudio/core/client/widget/LaunchShinyGadgetDialog.java

```diff
@@ -0,0 +1,15 @@
+package org.rstudio.core.client.widget;
+
+import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.Widget;
+
+public class LaunchShinyGadgetDialog extends ModalDialogBase
```

Comment:
When implementing this you'll want to be sure to do the `ensureStylesInjected` thing to make sure it properly centers on first show.